cmd/cli/config/auto.go

```diff
@@ -101,7 +104,16 @@ func newAutoResourceCmd() *cobra.Command {
 			if err := settings.validate(); err != nil {
 				return err
 			}
-			return autoConfig(cmd.Context(), settings)
+			// create or open a repo
+			repoPath, err := cfg.RepoPath()
+			if err != nil {
+				return err
+			}
+			_, err = setup.SetupBacalhauRepo(repoPath, cfg)
+			if err != nil {
+				return err
+			}
```

Comment:
> why do you need to initialize a repo

Since we no longer create a repo in the root command, we must create on here if it's not present. This is because this CLI method modifies the config file in a repo, creating a config file when one isn't present.

> You are already passing a config to newAutoResourceCmd. How was that initialized?

As was done prior to this change - [the config instance is created with default values](https://github.com/bacalhau-project/bacalhau/pull/3959/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R88)
see `pkg/config/config.go`
```go
func New(opts ...Option) *config {
	c := &config{
		user:       viper.New(),
		system:     viper.New(),
		defaultCfg: configenv.Production,
	}
	for _, opt := range opts {
		opt(c)
	}
	c.user.SetEnvPrefix(environmentVariablePrefix)
	c.user.SetTypeByDefaultValue(inferConfigTypes)
	c.user.AutomaticEnv()
	c.user.SetEnvKeyReplacer(environmentVariableReplace)
	c.setDefault(c.defaultCfg)
	return c
}
```

> and how initializing the repo will change the config?

This is slightly different than before:
Prior to this change, the config instance [set paths on itself](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/config/config.go#L69) that the [repo would be configured based on](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/repo/util.go#L21).
Now when initializing the repo, it modifies the config by setting paths on the config instance via the method [`EnsureRepoPathsConfigured`](https://github.com/bacalhau-project/bacalhau/pull/3959/files#diff-51b4dbd635889500fad1dac99dd1fbd3ecff53680cd5bad57f1f05a09fa9afe5R242). 
This change was implemented in this manner because we discussed removing the option for users to set all paths on the configurations themselves that fell within the repo. Instead, the repository now manages the paths it sets, with the top-level path being the repository path, which users are still allowed to set.